### PR TITLE
feat(ingest): working with multiple bigquery projects

### PR DESCRIPTION
**Pull request by anshbansal**
Able to get profiling results for a GCP project where we only have view rights
image

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/ge_profiling_config.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ class GEProfilingConfig(ConfigModel):
     partition_profiling_enabled: bool = Field(default=True, description="")
     bigquery_temp_table_schema: Optional[str] = Field(
         default=None,
-        description="On bigquery for profiling partitioned tables needs to create temporary views. You have to define a schema where these will be created. Views will be cleaned up after profiler runs. (Great expectation tech details about this (https://legacy.docs.greatexpectations.io/en/0.9.0/reference/integrations/bigquery.html#custom-queries-with-sql-datasource).",
+        description="On bigquery for profiling partitioned tables needs to create temporary views. You have to define a dataset where these will be created. Views will be cleaned up after profiler runs. (Great expectation tech details about this (https://legacy.docs.greatexpectations.io/en/0.9.0/reference/integrations/bigquery.html#custom-queries-with-sql-datasource).",
```
Comment (treff7es) [inline metadata-ingestion/src/datahub/ingestion/source/ge_profiling_config.py:114]: nice

File: metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py [hunk 2]
```diff
@@ -955,4 +967,15 @@ def _get_ge_dataset(
...
             },
         )
+        if platform is not None and platform == "bigquery":
+            # This is done as GE makes the name as DATASET.TABLE
+            # but we want it to be PROJECT.DATASET.TABLE instead for multi-project setups
+            logger.debug(f"Setting table name to be {pretty_name}")
+            batch._table = sa.text(pretty_name)
+            name_parts = pretty_name.split(".")
+            if len(name_parts) != 3:
+                logger.error(
```
Comment (treff7es) [inline metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py:977]: Is it fine we just log on this error?

File: metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py [hunk 3]
```diff
@@ -332,8 +334,12 @@ def __init__(self, config, ctx):
         self.partition_info: Dict[str, str] = dict()
         atexit.register(cleanup, config)
 
-    def get_db_name(self, inspector: Inspector = None) -> str:
-        if self.config.project_id:
+    def get_db_name(
+        self, inspector: Inspector = None, for_sql_queries: bool = True
```
Comment (treff7es) [inline metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py:338]: Maybe some comment would be nice about the `for_sql_queries` parameter

File: metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py [hunk 4]
```diff
@@ -359,14 +362,11 @@ def _compute_big_query_lineage(self) -> None:
         )
         logger.debug(f"lineage metadata is {self.lineage_metadata}")
 
-    def _compute_bigquery_lineage_via_gcp_logging(
-        self, lineage_client_project_id: Optional[str]
-    ) -> None:
+    def _compute_bigquery_lineage_via_gcp_logging(self) -> None:
+        project_id = self.get_db_name()
```
Comment (treff7es) [inline metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py:366]: Maybe rename `project_id` to `gcp_logging_project_id`?

**Review by treff7es — APPROVED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py (hunk 2 above)
Comment (anshbansal) [reply metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py:977]: Noticed there were some cases where it wasn't three for some people in OSS. Don't want to break until we understand why.

File: metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py (hunk 3 above)
Comment (anshbansal) [reply metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py:338]: added

**Pull request merged**